netctl/netctl.go

```diff
@@ -1000,6 +1008,127 @@ func dumpInspectList(ctx *cli.Context, list interface{}) {
 	os.Stdout.WriteString("\n")
 }
 
+// TODO: we should just use auth_proxy's proxy.loginReq, but it needs to be exported and Godep'd
+// TODO: even better, just export a function which takes username/pass and returns the JSON string
+type loginPayload struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+func login(ctx *cli.Context) {
+	if len(ctx.Args()) != 0 {
+		errExit(ctx, exitHelp, "More arguments than required", true)
+	}
+
+	// ensure the --netmaster flag starts with https
+	if !strings.HasPrefix(baseURL(ctx), "https://") {
+		err := errors.New("login requires a https auth_proxy URL")
+		errExit(ctx, exitInvalid, err.Error(), true)
+	}
+
+	// read username from stdin
+	fmt.Print("Username: ")
+
+	reader := bufio.NewReader(os.Stdin)
+	username, err := reader.ReadString('\n')
```

Comment:
It would be neat to have a separate function which reads the username and password ?